config/dynamic.c

```diff
@@ -291,6 +292,171 @@ cleanup_dy(struct dynamic_config *dy_conf)
 		rte_free(dy_conf->server_path);
 		dy_conf->server_path = NULL;
 	}
+
+	destroy_mailbox(&dy_conf->mb);
+}
+
+static void
+process_return_message(lua_State *l, struct dynamic_config *dy_conf,
+	int num_succ_sent_inst)
+{
+	int num_gt_messages = 0;
+	size_t reply_len = 0;
+	char reply_msg[MSG_MAX_LEN];
+	int num_cmd_acc = 0;
+	struct dy_cmd_entry *dy_cmds[dy_conf->mailbox_burst_size];
+
+	/* Wait for all GT instances to synchronize. */
+	while (rte_atomic16_read(&dy_conf->num_returned_instances)
+			< num_succ_sent_inst)
+		rte_pause();
+
+	while (num_cmd_acc < num_succ_sent_inst) {
+		int i;
+		/* Load a set of commands from its mailbox ring. */
+		int num_cmd = mb_dequeue_burst(&dy_conf->mb,
+			(void **)dy_cmds, dy_conf->mailbox_burst_size);
+
+		for (i = 0; i < num_cmd; i++) {
+			struct dy_cmd_entry *entry = dy_cmds[i];
+			switch (entry->op) {
+				case GT_UPDATE_POLICY_RETURN: {
+
+					if (dy_conf->gt == NULL) {
+						DYC_LOG(ERR, "The command operation %u requires that the server runs as Grantor\n",
+							entry->op);
+						break;
+					}
+
+					if (reply_len + entry->u.gt.length >= MSG_MAX_LEN)
+						DYC_LOG(ERR, "The aggregated return message from GT blocks is too long\n");
+					else {
+						strncpy(reply_msg + reply_len,
+							entry->u.gt.return_msg, entry->u.gt.length);
+						reply_len += entry->u.gt.length;
```

Comment:
We cannot use `strncopy()` because we are assuming that the string is binary, so it can have a `\0` in the middle. Use `rte_memcpy()` instead.